llvm/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.cpp

```diff
@@ -1158,14 +1158,23 @@ unsigned getMaxNumVGPRs(const MCSubtargetInfo *STI, unsigned WavesPerEU) {
   return std::min(MaxNumVGPRs, AddressableNumVGPRs);
 }
 
-unsigned getNumVGPRBlocks(const MCSubtargetInfo *STI, unsigned NumVGPRs,
-                          std::optional<bool> EnableWavefrontSize32) {
-  NumVGPRs = alignTo(std::max(1u, NumVGPRs),
-                     getVGPREncodingGranule(STI, EnableWavefrontSize32));
-  // VGPRBlocks is actual number of VGPR blocks minus 1.
-  return NumVGPRs / getVGPREncodingGranule(STI, EnableWavefrontSize32) - 1;
+static unsigned getNumBlocks(unsigned NumVGPRs, unsigned Granule) {
```

Comment:
Add something about VGPRs here? getNumBlocks sounds too vague.

Reply:
Now that you mention it, there's nothing VGPR specific here, since we're passing in both the number of registers and the granule. In fact, I should probably use this for `getNumSGPRBlocks` too. Maybe `getNumRegisterBlocks` would be best?

Reply:
Yes, I thought so too, it is not VGPR specific. Maybe even something like getGranulatedNumRegisterBlocks. Plus rename the first argument, it is not VGPR number either, just register number.